File: src/com/nerdscorner/android/plugin/github/domain/gh/GHReleaseWrapper.java
package com.nerdscorner.android.plugin.github.domain.gh;

import org.kohsuke.github.GHRelease;

import java.net.URL;

public class GHReleaseWrapper extends Wrapper {
    private GHRelease ghRelease;

    public GHReleaseWrapper(GHRelease ghRelease) {
        this.ghRelease = ghRelease;
    }

    public String getUrl() {
        return ghRelease.getUrl().toString();
    }

    public URL getFullUrl() {
        return ghRelease.getHtmlUrl();
    }

    public GHRelease getGhRelease() {
        return ghRelease;
    }

    @Override
    public String toString() {
        return ghRelease.getName();
    }

    @Override
    public int compare(Wrapper other) {
        if (other instanceof GHReleaseWrapper) {
            GHReleaseWrapper otherWrapper = (GHReleaseWrapper) other;
            try {
                return otherWrapper.ghRelease.getUpdatedAt().compareTo(ghRelease.getUpdatedAt());
            } catch (Exception ignored) {
            }
        }
        return 0;
    }
}

File: src/com/nerdscorner/android/plugin/github/domain/gh/GHReleaseWrapper.kt
package com.nerdscorner.android.plugin.github.domain.gh

import com.nerdscorner.android.plugin.utils.Strings
import org.kohsuke.github.GHRelease

import java.net.URL
import java.text.SimpleDateFormat

class GHReleaseWrapper(val ghRelease: GHRelease) : Wrapper() {

    val url: String
        get() = ghRelease.url.toString()

    val fullUrl: URL
        get() = ghRelease.htmlUrl

    val publishedAt = try {
        SimpleDateFormat(Strings.DATE_FORMAT).format(ghRelease.published_at)
    } catch (e: Exception) {
        null
    }

    override fun toString(): String {
        return ghRelease.name
    }

    override fun compare(other: Wrapper): Int {
        if (other is GHReleaseWrapper) {
            try {
                return other.ghRelease.updatedAt.compareTo(ghRelease.updatedAt)
            } catch (_: Exception) {
            }
        }
        return 0
    }
}
